salt-mode.el

```diff
@@ -82,6 +82,50 @@ suitable for spellchecking."
 
 (put 'salt-mode 'flyspell-mode-predicate #'salt--flyspell-predicate)
 
+(defun salt--state-module-at-point ()
+  "Get the state module at point, either pkg or pkg.installed, or return nil."
+  (save-excursion
+    (when (looking-back ":" 1)
+      (backward-char))
+    (skip-chars-forward " ")
+    (let* (start
+           end
+           (module (progn (skip-chars-backward "_.a-z0-9")
+                          (setq start (point))
+                          (forward-char)
+                          (skip-chars-forward "_.a-z0-9")
+                          (setq end (point))
+                          (string-trim (buffer-substring-no-properties start end)))))
+      (if (string= "" module) nil module))))
+
+(defun salt--doc-read-arg ()
+  "Get the argument for interactively calling `salt-open-doc'"
+  (let* ((default (salt--state-module-at-point))
+         (prompt (if default
+                     (format "Open salt doc (%s): " default)
+                   "Open salt doc, e.g. file.managed: "))
+         (word (if (or current-prefix-arg (not default))
+                   (completing-read prompt nil nil nil nil nil default)
+                 default)))
+    (list word)))
+
+(defun salt-open-doc (module)
```

Comment:
I would call it `salt-mode-browse-doc` or something to indicate that it opens in a browser, because eventually I also want a function that tries to open the documentation directly in another buffer.